src/dsl.jl

```diff
@@ -914,68 +931,61 @@ end
 # Function for creating a Reaction structure (used by the @reaction macro).
 function make_reaction(ex::Expr)
 
-    # Handle interpolation of variables
+    # Handle interpolation of variables in the input.
     ex = esc_dollars!(ex)
 
-    # Parses reactions, species, and parameters.
+    # Parses reactions. Extracts species and parameters within it.
     reaction = get_reaction(ex)
-    species, parameters = extract_species_and_parameters!([reaction], [])
+    species, parameters = extract_sps_and_ps([reaction], [])
 
-    # Checks for input errors.
+    # Checks for input errors. Needed here but not in `@reaction_network` as `ReactionSystem` perform this check but `Reaction` don't.
     forbidden_symbol_check(union(species, parameters))
```

Comment:
For `@reaction_network` this check is not needed as `ReactionSystem` does a similar one. We might consider whether `Reaction` should also perform this type of check (and we remove it from here entirely.

E.g. this works:
```julia
@parameters Γ
@species X(t)
Reaction(Γ, [X], [])
```
while this errors:
```julia
@reaction Γ, X --> 0
```